Sources/TuistGenerator/Generator/TargetGenerator.swift

```diff
@@ -156,7 +156,7 @@ final class TargetGenerator: TargetGenerating {
                 let nativeTarget = nativeTargets[targetSpec.name]!
                 let nativeDependency = nativeTargets[dependency.target.name]!
                 let pbxTargetDependency = try nativeTarget.addDependency(target: nativeDependency)
-                pbxTargetDependency?.applyCondition(condition, applicableTo: dependency.target)
+                pbxTargetDependency?.applyCondition(condition, applicableTo: targetSpec)
```

Comment:
this is already being accounted for in https://github.com/tuist/tuist/pull/5690

Reply:
Thanks for the info :) I will update this PR once https://github.com/tuist/tuist/pull/5690 is merged

Reply:
Merged `main` now